core/src/main/java/org/apache/servicecomb/core/handler/ShutdownHookHandler.java

```diff
@@ -39,13 +40,16 @@
 
   private final AtomicLong responseCounter = new AtomicLong(0);
 
-  private final int timeout = 600;
-
-  private final int period = 10;
-
   private volatile boolean shuttingDown = false;
 
+  public final Semaphore ALL_INVOCATION_FINISHED = new Semaphore(1);
+
   private ShutdownHookHandler() {
+    try {
+      ALL_INVOCATION_FINISHED.acquire();
```

Comment:
I think it's better to make semaphore private and add a method like waitInvocationFinish.  And in initialization, acquire is not required and shutdownhook can be removed too.